File: Euclid/Network/Streams/GameAddress.cs
namespace Euclid.Network.Streams
{
    public class GameAddress
    {
        #region Fields

        public string IpAddress { get; }
        public int Port { get; }
        public int RoomId { get; }

        #endregion

        #region Constructors

        public GameAddress(string ipAddress, int port, int roomId = 0)
        {
            IpAddress = ipAddress;
            Port = port;
            RoomId = roomId;
        }

        #endregion
    }
}

File: euclid/src/main/java/euclid/network/streams/GameAddress.java
package euclid.network.streams;

public final class GameAddress {

    private final String ipAddress;
    private final int port;
    private final int roomId;

    public GameAddress(String ipAddress, int port) {
        this(ipAddress, port, 0);
    }

    public GameAddress(String ipAddress, int port, int roomId) {
        this.ipAddress = ipAddress;
        this.port = port;
        this.roomId = roomId;
    }

    public String getIpAddress() {
        return ipAddress;
    }

    public int getPort() {
        return port;
    }

    public int getRoomId() {
        return roomId;
    }
}
